src/bin/casr-python.rs

```diff
@@ -0,0 +1,282 @@
+
+    // Create report.
+    let mut report = CrashReport::new();
+    report.executable_path = argv[0].to_string();
+    report.proc_cmdline = argv.join(" ");
+    let _ = report.add_os_info();
+
+    // Get python report.
+    let python_stderr_list: Vec<String> =
+        python_stderr.split('\n').map(|l| l.to_string()).collect();
+
+    let re = Regex::new(r"==\d+==\s*ERROR: (LeakSanitizer|AddressSanitizer|libFuzzer):").unwrap();
+    if python_stderr_list.iter().any(|line| re.is_match(line)) {
+        let python_stdout = String::from_utf8_lossy(&python_result.stdout);
+        let python_stdout_list: Vec<String> =
+            python_stdout.split('\n').map(|l| l.to_string()).collect();
+
+        if let Some(report_start) = python_stdout_list
+            .iter()
+            .position(|line| line.contains("Uncaught Python exception: "))
+        {
+            // Set python report in casr report.
+            if let Some(report_end) = python_stdout_list.iter().rposition(|s| !s.is_empty()) {
+                let report_end = report_end + 1;
+                if report_end > python_stdout_list.len() {
+                    bail!("Corrupted output: can't parse stdout end");
+                }
+                report.python_report = Vec::from(&python_stdout_list[report_start..report_end]);
+
+                // Get exception from python report.
+                if let Ok(exception) = python_exception(&report.python_report[1]) {
+                    report.execution_class = exception;
+                }
+
+                // Get stack trace from python report.
+                let first = report
+                    .python_report
+                    .iter()
+                    .position(|line| line.starts_with("Traceback "));
+                if first.is_none() {
+                    bail!("Couldn't find traceback in python report");
+                }
+
+                // Stack trace is splitted by empty line.
+                let first = first.unwrap();
+                let last = report
+                    .python_report
+                    .iter()
+                    .skip(first)
+                    .rposition(|s| !s.is_empty());
+                if last.is_none() {
+                    bail!("Couldn't find traceback end in python report");
+                }
+                let last = last.unwrap();
+
+                let re = Regex::new(
+                    r#"(File ".+", line [\d]+, in .+|\[Previous line repeated (\d+) more times\])"#,
+                )
+                .unwrap();
+                report.stacktrace = report.python_report[first..first + last]
+                    .iter()
+                    .rev()
+                    .map(|s| s.trim().to_string())
+                    .filter(|s| re.is_match(s))
+                    .collect::<Vec<String>>();
+            } else {
+                bail!("Corrupted output: can't find stdout end");
+            }
+        } else {
+            // Call casr-san
```

Comment:
this code repeats twice, we may create a function

Reply:
Done